code/datums/elements/point_of_interest.dm

```diff
@@ -5,9 +5,21 @@
 /datum/element/point_of_interest/Attach(datum/target)
 	if (!isatom(target))
 		return ELEMENT_INCOMPATIBLE
-	GLOB.poi_list += target
+
+	// Do not ever attach to players in the lobby.
+	if(isnewplayer(target))
+		return ELEMENT_INCOMPATIBLE
+
+	// Do not ever attach to stealthmins. This is not perfect, as mobs can ascend to stealthmin status later.
+	// SSpois steps in at that point to make sure the stealthmin is not considered a POI.
+	if(ismob(target))
+		var/mob/target_mob = target
+		if(target_mob.client?.holder?.fakekey)
+			return ELEMENT_INCOMPATIBLE
```

Comment:
I am still wary about runtiming here.

Reply:
This should never runtime because target_mob is never null here thanks to the ismob check.

Would you prefer I do `if(target_mob.client && target_mob.client.holder && target_mob.client.holder.fakekey)` instead of the shorthand `?.` null checks?

If you're talking about this runtiming because of ELEMENT_INCOMPATIBLE, see above. It's an intended CRASH state for attempting to attach elements to things that shouldn't have the element attached along with debugging output to track down the cause and fix it.

Reply:
I'm talking about the CRASH, though if you're already preventing these everywhere else then it's fine.

Can you document both of these in the element itself? Why you're runtiming there?

Reply:
This is a bit different in the latest version. The only check left is against isnewplayer() which is more documented and stealthmins are managed through validation as part of the point_of_interest datum and SSpoints_of_interest running validation.